packages/shared/lib/src/sdk/args.rs

```diff
@@ -0,0 +1,611 @@
+use std::{path::PathBuf, str::FromStr};
+
+use namada::core::borsh::{BorshDeserialize, BorshSerialize};
+use namada::core::ibc::core::host::types::identifiers::{ChannelId, PortId};
+use namada::tendermint_rpc;
+use namada::tx::data::GasLimit;
+use namada::{
+    address::Address,
+    chain::ChainId,
+    ethereum_events::EthAddress,
+    key::common::PublicKey,
+    masp::TransferSource,
+    sdk::args::{self, InputAmount, TxExpiration},
+    token::{Amount, DenominatedAmount, NATIVE_MAX_DECIMAL_PLACES},
+};
+use wasm_bindgen::JsError;
+
+#[derive(BorshSerialize, BorshDeserialize)]
+#[borsh(crate = "namada::core::borsh")]
+pub struct RevealPkMsg {
+    public_key: String,
+}
+
+impl RevealPkMsg {
+    pub fn new(public_key: String) -> RevealPkMsg {
+        RevealPkMsg {
+            public_key
+        }
+    }
+}
+
+#[derive(BorshSerialize, BorshDeserialize)]
+#[borsh(crate = "namada::core::borsh")]
+pub struct WrapperTxMsg {
+    token: String,
+    fee_amount: String,
+    gas_limit: String,
+    chain_id: String,
+    public_key: Option<String>,
+    memo: Option<String>,
+}
+
+impl WrapperTxMsg {
+    pub fn new(
```

Comment:
I think you can already create instances of `WrapperTxMsg` like this just by defining the struct and leaving out the `new` method (but it's been a while since I wrote any Rust so I could be wrong about that :smile:).

Reply:
the problem is is that I moved it into `tx.rs`, so anything outside of that file must use a public accessor/setter. Those fields are all treated as private for these wasm-bindgen structs, and it's annoying, but that's why I added it!

Reply:
Ah, I see! I thought I must be missing something :smile: